gravitee-node-secrets/gravitee-node-secrets-service/src/main/java/io/gravitee/node/secrets/service/conf/GraviteeConfigurationSecretResolverDispatcher.java

```diff
@@ -0,0 +1,182 @@
+package io.gravitee.node.secrets.service.conf;
+
+import io.gravitee.common.util.EnvironmentUtils;
+import io.gravitee.node.api.secrets.SecretManagerConfiguration;
+import io.gravitee.node.api.secrets.SecretProvider;
+import io.gravitee.node.api.secrets.errors.SecretManagerConfigurationException;
+import io.gravitee.node.api.secrets.errors.SecretManagerException;
+import io.gravitee.node.api.secrets.errors.SecretProviderNotFoundException;
+import io.gravitee.node.api.secrets.model.*;
+import io.gravitee.node.api.secrets.util.ConfigHelper;
+import io.gravitee.node.secrets.plugins.SecretProviderPluginManager;
+import io.gravitee.node.secrets.service.AbstractSecretProviderDispatcher;
+import io.reactivex.rxjava3.core.Maybe;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+
+/**
+ * @author Benoit BORDIGONI (benoit.bordigoni at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+public class GraviteeConfigurationSecretResolverDispatcher extends AbstractSecretProviderDispatcher {
```

Comment:
Can you explain why you had to create this Dispatcher and do not use the SecretPluginManager directly instead ?

Reply:
The plugin class is the factory, not the service. Thus, plugin manager contains factories only. I needed the SecretProvider (service) created from that factory in order to call them.

Reply:
Ok I get it, it looks like what has been done for connectors.